### PR TITLE
Refactor e2e tests to use kueue

### PR DESCRIPTION
**Pull request by ChughShilpa**
# Issue link
<https://issues.redhat.com/browse/RHOAIENG-5239>

# What changes have been made
Refactored e2e tests to use kueue and not mcad. 
Added steps to create kueue resources such as cluster queue, resource flavour and local queue.

# Verification steps
Execute e2e tests following https://github.com/project-codeflare/codeflare-sdk/blob/main/docs/e2e.md 

## Checks
- [X] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: tests/e2e/kueue_resources_setup.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+cat <<EOF | kubectl apply --server-side -f - 
+    apiVersion: kueue.x-k8s.io/v1beta1
+    kind: ClusterQueue
+    metadata:
+        name: $name
+    spec:
+      namespaceSelector: {}
+      resourceGroups:
+      - coveredResources: ["cpu", "memory", "pods", "nvidia.com/gpu"]
```
Comment (sutaakar) [inline tests/e2e/kueue_resources_setup.sh:18]: IMHO there is no need to cover number of pods.

**Comment by sutaakar** [comment]
`tests/e2e/mnist_raycluster_sdk_test.py` seems to be a duplication of one test case in `tests/e2e/mnist_raycluster_sdk_oauth_test.py` . IMHO `mnist_raycluster_sdk_test.py` can be removed.

**Review by Fiona-Waters — CHANGES_REQUESTED** [review]
Just a couple of nitpicks

**Comment by ChughShilpa** [comment]
> `tests/e2e/mnist_raycluster_sdk_test.py` seems to be a duplication of one test case in `tests/e2e/mnist_raycluster_sdk_oauth_test.py` . IMHO `mnist_raycluster_sdk_test.py` can be removed.

Removed `mnist_raycluster_sdk_test.py` 
Updated dedicated test for kind cluster `assert_jobsubmit_withoutlogin_kind`

File: tests/e2e/mnist_raycluster_sdk_oauth_test.py [hunk 2]
```diff
@@ -47,19 +47,21 @@ def run_mnist_raycluster_sdk_oauth(self):
...
         )
 
         cluster.up()
-        self.assert_appwrapper_exists()
 
         cluster.status()
 
-        cluster.wait_ready()
+    # Uncomment cluster.wait_ready() once the issue https://issues.redhat.com/browse/RHOAIENG-3981 is resolved
+    #    cluster.wait_ready
```
[the remark this answers is not in the document]
Comment (ChughShilpa) [reply tests/e2e/mnist_raycluster_sdk_oauth_test.py]: Added cluster.wait_ready again as the issue has been fixed

**Comment by Fiona-Waters** [comment]
Ran test against an openshift cluster and it passes as expected.
/lgtm

File: docs/e2e.md [hunk 3]
```diff
@@ -10,8 +10,13 @@ Pre-requisite for KinD clusters: please add in your local `/etc/hosts` file `127
   ```
   make kind-e2e
   export CLUSTER_HOSTNAME=kind
-  make deploy -e IMG=quay.io/project-codeflare/codeflare-operator:v1.1.0
   make setup-e2e
+
+  # For running tests locally on Kind cluster, we need to disable oauth (which is enabled by default) by setting the boolean value as false
+  [Disable oauth](https://github.com/project-codeflare/codeflare-operator/blob/main/main.go#L117)
```
Comment (sutaakar) [inline docs/e2e.md:16]: This can be done by modifying `codeflare-operator-config` ConfigMap, disabling `rayDashboardOAuthEnabled` and restarting CFO.

**Comment by Fiona-Waters** [comment]
I have reviewed again with most recent changes on an openshift cluster - test passed as expected.
/lgtm

**Review by sutaakar — APPROVED** [review]
/lgtm
PR check errors are probably caused by RBAC issues, will be investigated in separate PR

**Pull request merged**